src/nvim/api/buffer.c

```diff
@@ -467,7 +467,7 @@ void buffer_insert(Buffer buffer,
                    ArrayOf(String) lines,
                    Error *err)
 {
-  buffer_set_line_slice(buffer, lnum, lnum, false, true, lines, err);
+  buffer_set_line_slice(buffer, lnum, lnum, true, false, lines, err);
```

Comment:
As I mentioned in https://github.com/neovim/neovim/pull/2846 I think `include_start`/`include_end` parameters shouldn't exist at all. Thanks for the bug fix in the meantime.